sdk/appconfiguration/app-configuration/src/internal/helpers.ts

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { ETagOption, ListConfigurationSettingsOptions, AppConfigurationGetKeyValuesOptionalParams } from '..';
+import { URLBuilder } from '@azure/core-http';
+import { isArray } from 'util';
+
+/**
+ * Formats the etag so it can be used with a if-match header
+ * @internal
+ * @ignore
+ */
+export function formatETagForMatchHeaders(objectWithEtag: ETagOption): string | undefined {
+  if (objectWithEtag.etag) {
+    return `"${objectWithEtag.etag}"`;
+  }
+
+  return undefined;
+}
+
+/**
+ * Transforms the keys/labels parameters in the listConfigurationSettings and listRevisions
+ * into the format the REST call will need.
+ * 
+ * @internal
+ * @ignore
+ */
+export function formatWildcards(
+  listConfigOptions?: ListConfigurationSettingsOptions
+): Pick<AppConfigurationGetKeyValuesOptionalParams, "key" | "label"> {
+  if (listConfigOptions == null) {
+    return {};
+  }
+
+  let key = undefined;
+
+  if (listConfigOptions.keys) {
+    // TODO: escape commas?
+    key = listConfigOptions.keys.join(",");
+  }
+
+  let label = undefined;
+
+  if (listConfigOptions.labels) {
+    label = listConfigOptions.labels.join(",");
+  }
+
+  return {
+    key,
+    label
+  };
+}
+
+/**
+ * Take the URL that gets returned from next link and extract the 'after' token needed
+ * to get the next page of results.
+ * @internal
+ * @ignore
+ */
+export function extractAfterTokenFromNextLink(nextLink: string) {
+  let parsedLink = URLBuilder.parse(nextLink);
+  let afterToken = parsedLink.getQueryParameterValue("after");
+
+  if (afterToken == null || isArray(afterToken)) {
```

Comment:
so if there are multiple `after` query params, we give up? I suppose there isn't really an obvious alternative, except maybe taking the first one...

Reply:
This is more that there shouldn't be multiple 'after' query parameters but the URLBuilder generically allows it, since it's valid for query parameters in general.

But just not for _this_ query parameter. :)